Abcbadq/sjsu_shop
# SJSU-1011, Shopping Cart

Story: Create a Shopping Cart
What I did:
Implemented count tracking for products in cart and an array of distinct products 
implemented increment and decrement product count in shopping cart
implemented delete all for a product(with any count >= 1) in cart
Implemented wiring to checkout via checkout button

*Installation needed on your local machine, cd to front_end under sjsu_shop:
$ npm install --save react react-dom 
$ npm install --save react-bootstrap

To Verify:
Suggestions for steps for testing:
1.	Click on shopping cart icon on the navbar: Shopping Cart is empty
2.	Click on Categories or search, click on add to cart buttons OR click on item name which directs to product detail page, and click on add to cart button
3.	Click on cart icon on navbar: Shopping cart should have correct count for each product
4.	Delete in any sequence, from front/middle/end of array, with any delete/decrement button and nothing breaks. Increment should still work.
5.	Click on checkout button, redirects user to checkout page and displays correct messages/product details. 
6.	Shopping cart modal should disappear when on checkout page
7.	Click on shopping cart icon, shopping cart modal should appear for user to easily edit products in shopping cart while on the checkout page.
8.	User makes edits on shopping cart modal. User clicks checkout, redirects user to checkout page and shopping cart modal should disappear again.
9.	If user empties cart, cart should become empty. 
10.	Repeat any combination of above steps and nothing should break.

How to get array of cart with products for other features etc. storing for user profile:
Replace YourOwnClassName, handle your own conditions (since this implementation allows for best code reuse, but will send data to connected js pages whenever reducer state changes) 

import { connect } from "react-redux";

render() {
    const cart = this.props.cart.shoppingCart;

const mapStateToProps = state => ({
  cart: state.cart
});

//connect to cartReducer to display items in cart
export default connect(
  mapStateToProps,
  {}
)(YourOwnClassName);
